### PR TITLE
Added MigrationSecuritySettings and CertificateHelper

### PR DESCRIPTION
**Pull request by roshanroutray**
Enabled HTTPS endpoint for migration REST API

### CODE REVIEW

File: src/Microsoft.ServiceFabric.Actors.KVSToRCMigration/CertificateHelper.cs [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+            foreach (var val in securitySettings.CertificateRemoteCommonNames.Split(','))
+            {
+                clientCertList.AddRange(FindMatchingCertificates(storeName, storeLocation, X509FindType.FindBySubjectName, val, includeExpiredCerts));
+            }
+
+            return clientCertList;
+        }
+
+        internal static List<X509Certificate2> FindMatchingCertificates(
+            string storeName,
```
Comment (pakunapamsft) [inline src/Microsoft.ServiceFabric.Actors.KVSToRCMigration/CertificateHelper.cs:56]: The logic in this file is equivalent to certificate manager in system.fabric.common namespace?

**Review by pakunapamsft — CHANGES_REQUESTED** [review]

File: src/Microsoft.ServiceFabric.Actors.KVSToRCMigration/CertificateHelper.cs (hunk 1 above)
Comment (roshanroutray) [reply src/Microsoft.ServiceFabric.Actors.KVSToRCMigration/CertificateHelper.cs:56]: yes

File: src/Microsoft.ServiceFabric.Actors.KVSToRCMigration/HttpCommunicationClient.cs [hunk 2]
```diff
@@ -42,5 +31,54 @@ public HttpCommunicationClient(string address)
...
+        {
+            var handler = new HttpClientHandler();
+
+            if (clientCertificates != null)
+            {
+                handler.ClientCertificateOptions = ClientCertificateOption.Manual;
+                handler.ClientCertificates.AddRange(clientCertificates);
+                handler.ServerCertificateCustomValidationCallback += (sender, certificate, chain, sslPolicyErrors) =>
+                {
+                    return CertificateHelper.IsValidRemoteCert(certificate, chain, securitySettings);
```
[the remark this answers is not in the document]
Comment (roshanroutray) [reply src/Microsoft.ServiceFabric.Actors.KVSToRCMigration/HttpCommunicationClient.cs:59]: This is the validation for certificate presented by server (KVS source) on client (RC target) side. So, on client side we will be validating the remotethumbprints

File: src/Microsoft.ServiceFabric.Actors.KVSToRCMigration/HttpCommunicationClient.cs [hunk 3]
```diff
@@ -42,5 +31,54 @@ public HttpCommunicationClient(string address)
         public string ListenerName { get; set; }
 
         public ResolvedServiceEndpoint Endpoint { get; set; }
+
+        private HttpClient GetHttpClient(Uri endpointUri, MigrationSecuritySettings securitySettings)
+        {
+            var certs = CertificateHelper.GetCertificates(securitySettings);
```
[the remark this answers is not in the document]
Comment (roshanroutray) [reply src/Microsoft.ServiceFabric.Actors.KVSToRCMigration/HttpCommunicationClient.cs:37]: yes. will do

File: src/Microsoft.ServiceFabric.Actors.KVSToRCMigration/HttpCommunicationClient.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (roshanroutray) [reply src/Microsoft.ServiceFabric.Actors.KVSToRCMigration/HttpCommunicationClient.cs:59]: keeping it as if in Fabric Transport setting. Lets discuss this if a separate config for Migration Client and Migration listener is needed.

**Comment by roshanroutray** [comment]
Resolved comments, Rebased and created new PR on develop
https://github.com/microsoft/service-fabric-services-and-actors-dotnet/pull/325

**Pull request closed without merging**